izumi-reflect/izumi-reflect/src/main/scala/izumi/reflect/macrortti/LightTypeTag.scala

```diff
@@ -104,6 +104,28 @@ abstract class LightTypeTag(
     LightTypeTag(ref.combine(argRefs), mergedBasesDB, mergedInheritanceDb)
   }
 
+  def asUnion(args: LightTypeTag*): LightTypeTag = {
+    def mergedBasesDB = LightTypeTag.mergeIDBs(basesdb, args.iterator.map(_.basesdb))
+
+    def mergedInheritanceDb = LightTypeTag.mergeIDBs(idb, args.iterator.map(_.idb))
+
+    LightTypeTag(
+      LightTypeTagRef.UnionReference(
+        args
+          .map(_.ref match {
+            case reference: AbstractReference =>
+              reference match {
+                case Lambda(input, output) => ???
```

Comment:
We need to throw something readable here.

Reply:
Ah yeah, did that. Copying tests from the Scala 2 tag tests to Shared (that I can) now, then I'll push :)